sdk/servicebus/service-bus/src/serviceBusAtomManagementClient.ts

```diff
@@ -288,40 +375,93 @@ export class ServiceBusManagementClient extends ServiceClient {
    * @param connectionString The connection string needed for the client to connect to Azure.
    * @param options ServiceBusManagementClientOptions
    */
-  constructor(connectionString: string, options?: ServiceBusManagementClientOptions) {
-    const connectionStringObj: any = parseConnectionString(connectionString);
+  constructor(connectionString: string, options?: ServiceBusManagementClientOptions);
+  /**
+   *
+   * @param fullyQualifiedNamespace The full namespace of your Service Bus instance which is
+   * likely to be similar to <yournamespace>.servicebus.windows.net.
+   * @param credential A credential object used by the client to get the token to authenticate the connection
+   * with the Azure Service Bus. See &commat;azure/identity for creating the credentials.
+   * If you're using an own implementation of the `TokenCredential` interface against AAD, then set the "scopes" for service-bus
+   * to be `["https://servicebus.azure.net//user_impersonation"]` to get the appropriate token.
+   * @param options ServiceBusManagementClientOptions
+   */
+  constructor(
+    fullyQualifiedNamespace: string,
+    credential: TokenCredential,
+    options?: ServiceBusManagementClientOptions
+  );
+
+  constructor(
+    fullyQualifiedNamespaceOrConnectionString1: string,
+    credentialOrOptions2?: TokenCredential | ServiceBusManagementClientOptions,
+    options3?: ServiceBusManagementClientOptions
+  ) {
+    if (isTokenCredential(credentialOrOptions2)) {
+      const requestPolicyFactories: RequestPolicyFactory[] = [];
+      if (options3 && options3.proxySettings) {
+        requestPolicyFactories.push(proxyPolicy(options3.proxySettings));
+      }
+      const serviceClientOptions: ServiceClientOptions = {
+        requestPolicyFactories: requestPolicyFactories
+      };
+      super(credentialOrOptions2, serviceClientOptions);
+      this.sasTokenProvider = credentialOrOptions2;
```

Comment:
How would this work... ?
`sasTokenProvider` is of type `SharedKeyCredential` whose `getToken()` method takes a different argument than the `getToken()` method of `TokenCredential`.

Reply:
Yeah, it did not work, I've just added this to make the build succeed.
I'm still figuring out the token credential overload.

Reply:
I am ok if we pull the token credential work out of this PR and have this focus on just adding the missing APIs